examples/tv-app/linux/include/channel/ChannelManager.cpp

```diff
@@ -167,12 +167,25 @@ bool ChannelManager::HandleChangeChannelByNumber(const uint16_t & majorNumber, c
     return channelChanged;
 }
 
-bool ChannelManager::HandleSkipChannel(const uint16_t & count)
+bool ChannelManager::HandleSkipChannel(const int16_t & count)
 {
-    // TODO: Insert code here
-    uint16_t newChannelIndex = static_cast<uint16_t>((count + mCurrentChannelIndex) % mChannels.size());
-    mCurrentChannelIndex     = newChannelIndex;
-    mCurrentChannel          = mChannels[mCurrentChannelIndex];
+    int32_t newChannelIndex = static_cast<int32_t>(count) + static_cast<int32_t>(mCurrentChannelIndex);
+    uint16_t channelsSize   = static_cast<uint16_t>(mChannels.size());
+
+    // handle larger than maximum -> channel size case
+    if (newChannelIndex >= channelsSize)
+    {
+        newChannelIndex = newChannelIndex % channelsSize;
+    }
+
+    // handle smaller than 0
+    if (newChannelIndex < 0)
+    {
+        newChannelIndex = channelsSize + (newChannelIndex % channelsSize);
```

Comment:
This does not do the right thing either.  Where did this formula come from?

Writing some basic tests would hopefully have caught this, by the way.  Please do that.

Reply:
Thanks for the review @bzbarsky-apple .

I've added tests in [TV_ChannelCluster.yaml](https://github.com/project-chip/connectedhomeip/pull/26501/files#diff-0dcf2e8985b200e8529993d599d3c8a206c6c4c17e59f66d770f118e131a65af).

Given that channel cluster by example has 4 channels, what would be the SkipChannel increase or decrease tests that you suggest are missing?

So far we have:
1. Skip Channel by 1. Assert current Channel value
2. Skip Channel by 123. Assert current Channel value
3. Skip Channel by -123. Assert current Channel value

Reply:
If channelsSize is 4, I would recommend starting with channel set to 3 and testing the following (again, starting at 3 for each test):

1. Skip up by 1
2. Skip up by 2
3. ....
4. Skip up by 8.
5. Skip down by 1
6. Skip dowm by 2
7. ...
8. Skip down by 8
9. Skip up by 32,767
10. Skip down by 32,768
11. Any other values of interest (123, -123, fine).

Reply:
Updated tests.